src/utils/storage/index.ts

```diff
@@ -1,13 +1,24 @@
-const initStorage = <T extends keyof StorageKey>(key: T, storage: Storage) => {
+import type { ProductDetail, ProductOption } from '@/types';
+
+interface StorageKey {
+  authToken?: string;
+  order?: OrderInfo | null;
+}
+
+interface OrderInfo {
+  product: ProductDetail & { selectedOption: ProductOption | null; quantity: number };
+}
+
+export const initStorage = <T extends keyof StorageKey>(key: T, storage: Storage) => {
   const storageKey = `${key}`;
 
   const get = (): StorageKey[T] => {
     const value = storage.getItem(storageKey);
-
-    return JSON.parse(value as string);
+    return value ? JSON.parse(value) : null;
   };
+
   const set = (value: StorageKey[T]) => {
-    if (value == undefined || value == null) {
+    if (value === undefined || value === null) {
```

Comment:
요렇게 작성하면 언디파인드랑 널 둘다 잡을 수 있습니다. 많이 쓰는 패턴입니다
```suggestion
    if (value == null) {
```